src/abstractdataframe/where.jl

```diff
@@ -0,0 +1,181 @@
+using DataFrames: normalize_selection, ColumnIndex
+# where allows a transformation specification without a target column name or a column
+
+_process_where_pair(i::Int, a::Pair{<:Any, <:Base.Callable}) = first(a) => last(a) => Symbol(:x, i)
```

Comment:
This `Base.Callable` dispatch is probably causing our latency, right? Good to see the implementation.

Reply:
this should compile pretty fast, but I will check (I can add `@nospecialize` if it causes latency)